crypto/cms/cms_enc.c

```diff
@@ -81,7 +81,7 @@ BIO *ossl_cms_EncryptedContent_init_bio(CMS_EncryptedContentInfo *ec,
 
     if (enc) {
         calg->algorithm = OBJ_nid2obj(EVP_CIPHER_CTX_get_type(ctx));
-        if (calg->algorithm == NULL) {
+        if (calg->algorithm == NULL || calg->algorithm == OBJ_nid2obj(NID_undef)) {
```

Comment:
Why not just compare calg->algorithm to &OBJ_undef instead of going through the extra call?

Reply:
do you mean compare to `&(nid_objs[NID_undef])` rather than use a function call?

I considered that briefly but decided against it b/c I did not find any examples in the code where elements of `nid_objs[]` are referenced directly (and I guess that is not a surprise b/c `nid_objs[]` is declared as `static`).

I could do this instead:

```C
        if (calg->algorithm == NULL || calg->algorithm->nid == NID_undef)
```

Okay?